### ISSUE TITLE
Turn trainer class from `Climate-Learing` into all-purpose ML model class

### ISSUE THREAD

**Issue by georgemilosh** [AlessandroLovo/general_purpose#5]
I think that trainer class can be made more convenient and useful for more people. The most convenient feature that could be helpful in many applications is how hyperparameters are treated and automatic folders spawned with `config.json` and `runs.json`files. The issue is that `Climate-Learning` is too bulky and probably hard to understand for most people. To get more adopters simpler class would be preferred. This is not a easy thing to do because it depends heavily on other functions in `Learn2_new.py` of `Climate-Learning/PLASIM`. So here is a set of desired features and steps:

- [ ] The class should be useful for anyone doing a parametric study and could be later inherited for downstream tasks
- [ ] Check that similar class does not already exist somewhere on GitHub. Also consider writing it from scratch.
- [ ] Trainer class stops being associated with a specific tasks of `Climate-Learning`: write a general class that has nothing to do with operations such as balancing, masking etc
- [ ] Ideally it would not depend on whether the user chooses to work with tensorflow or pytorch
- [ ] Make sure that the routines are as simple as possible. Basically try to reduce complexity of the code while prioritising utility
- [ ] Good documentation is a plus.

**Comment by AlessandroLovo** [comment]
[This](https://pytorch.org/tutorials/beginner/hyperparameter_tuning_tutorial.html) might be sort of related

**Comment by georgemilosh** [comment]
Indeed. I am not sure they natively support k-fold cross-validation, although certainly a user can build on top of that.
